### PR TITLE
Va iir 369 post911 gi bill statuses controller new version

### PR DESCRIPTION
**Pull request by AdamKing0126**
## Post-9/11 GI Bill Statement of Benefits

This PR is the third and final in a series with the end goal being the replacement of a network call to EVSS with one to Lighthouse.

[15642](https://github.com/department-of-veterans-affairs/vets-api/pull/15642) created the service that handles the network call.
[15717](https://github.com/department-of-veterans-affairs/vets-api/pull/15717) added functionality to prevent the request being made if it's during scheduled downtime of the external service.

This PR creates a controller at the `v1` namespace, so that front end clients can make requests to `vets-api` which will result in data being retrieved from Lighthouse, instead of EVSS.

EVSS is slated to be decommissioned in Sept 2024, so all requests must be transitioned to Lighthouse.

**Note**: Delete the description statements, complete each step. **None are optional**, but can be justified as to why they cannot be completed as written. Provide known gaps to testing that may raise the risk of merging to production.

## Summary

- *This work is behind a feature toggle (flipper): **NO***
This work creates a new api endpoint in the `v1` namespace for front end clients to access.  We will handle feature toggling on the front end client.
- *(Summarize the changes that have been made to the platform)*
A new controller and `v1` namespace endpoint that requests Gi Bill Statement of Benefits information from Lighthouse
- *(If bug, how to reproduce)*
Not a bug
- *(What is the solution, why is this the solution?)*
Creating a new endpoint allows us to iterate on new functionality for the front end clients without impacting existing users.  It also acts as a feature toggle, so we can roll out the change over time.
- *(Which team do you work for, does your team own the maintenance of this component?)*
I work for VA-IIR and we are the current owner of this product
- *(If introducing a flipper, what is the success criteria being targeted?)*
Veterans can successfully request their Post-9/11 GI Bill Statement of Benefits using the new api endpoint.

## Related issue(s)

- [ticket](https://app.zenhub.com/workspaces/va-iir-6508c0bd79e64e0fb5855caf/issues/gh/department-of-veterans-affairs/va-iir/369)

## Testing done

- [x] *New code is covered by unit tests*
- *Describe what the old behavior was prior to the change*
- *Describe the steps required to verify your changes are working as expected. Exclusively stating 'Specs run' is NOT acceptable as appropriate testing*
modify `config/test.yml` with proper credentials to access the lighthouse sandbox environment, and make network calls to the controller using the tests.
- *If this work is behind a flipper:*
  - *Tests need to be written for both the flipper on and flipper off scenarios. [Docs](https://depo-platform-documentation.scrollhelp.site/developer-docs/feature-toggles-guide#Featuretogglesguide-Backendexample).*
  - *What is the testing plan for rolling out the feature?*

## Screenshots
_Note: Optional_

## What areas of the site does it impact?
*(Describe what parts of the site are impacted and*if*code touched other areas)*
The site does not yet access this endpoint, but once front end development is complete, the "Check your Benefits" tool [here](https://staging.va.gov/education/gi-bill/post-9-11/ch-33-benefit/) will call the new endpoint and handle the results.

## Acceptance criteria

- [x]  I added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [x]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [x]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

## Requested Feedback

(OPTIONAL)_What should the reviewers know in addition to the above. Is there anything specific you wish the reviewer to assist with. Do you have any concerns with this PR, why?_

This is a relatively large PR, but most of the LOC comes from two VCR cassettes.

### CODE REVIEW

File: spec/controllers/v1/post911_gi_bill_statuses_controller_spec.rb [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+      expect(response).to have_http_status(:not_found)
+      json_response = JSON.parse(response.body)
+      expect(json_response['error']['title']).to eq('Not Found')
+      expect(json_response['error']['detail']).to eq('Icn not found.')
+    end
+  end
+
+  context 'outside working hours' do
+    # midnight
+    before { Timecop.freeze(tz.parse('2nd Feb 1993 00:00:00')) }
```
Comment (AdamKing0126) [inline spec/controllers/v1/post911_gi_bill_statuses_controller_spec.rb:53]: 🐇🥚

**Comment by AdamKing0126** [comment]
I think these CODEOWNERS errors are popping up because I've set this PR to merge into a non-master branch.  Should resolve on its own.

**Review by khenson-oddball — APPROVED** [review]
LGTM 👍

**Review by ryan-mcneil — APPROVED** [review]
:shipit:

**Pull request merged**